src/main/java/runhistoryplus/patches/ScoreBreakdownRunHistoryPatch.java

```diff
@@ -0,0 +1,176 @@
+            if (stat.label == null) {
+                break;
+            }
+            scoreBreakdown.add(stat.label + ": " + stat.value);
+        }
+    }
+
+
+    @SpirePatch(clz = CardCrawlGame.class, method = SpirePatch.CONSTRUCTOR)
+    public static class ScoreBreakdownField {
+        @SpireRawPatch
+        public static void addScoreBreakdownField(CtBehavior ctBehavior) throws NotFoundException, CannotCompileException {
+            CtClass runData = ctBehavior.getDeclaringClass().getClassPool().get(RunData.class.getName());
+            String fieldSource = "public java.util.List score_breakdown;";
+            CtField field = CtField.make(fieldSource, runData);
+            runData.addField(field);
+        }
+    }
+
+    @SpirePatch(clz = DeathScreen.class, method = "createGameOverStats")
+    public static class DeathScreenPatch {
+        @SpirePostfixPatch
+        public static void logScoreBreakdown(DeathScreen _instance, ArrayList<GameOverStat> ___stats) {
+            createScoreBreakdown(___stats);
+        }
+    }
+
+    @SpirePatch(clz = DeathScreen.class, method = SpirePatch.CONSTRUCTOR)
+    public static class InsertEarlyStatCalcDeathPatch {
+        @SpireInsertPatch(locator = Locator.class)
+        public static void insertEarlyStatCalcDeath(DeathScreen __instance) {
+            // both methods are called twice now, but this shouldn't cause any problems
+            ReflectionHacks.privateMethod(GameOverScreen.class, "calculateUnlockProgress").invoke(__instance);
+            ReflectionHacks.privateMethod(DeathScreen.class, "createGameOverStats").invoke(__instance);
+        }
+
+        public static class Locator extends SpireInsertLocator {
+            public int[] Locate(CtBehavior ctMethodToPatch) throws CannotCompileException, PatchingException {
+                Matcher matcher = new Matcher.MethodCallMatcher(MusicMaster.class, "playTempBgmInstantly");
+                return LineFinder.findInOrder(ctMethodToPatch, new ArrayList<Matcher>(), matcher);
+            }
+        }
+    }
+
+    @SpirePatch(clz = VictoryScreen.class, method = "createGameOverStats")
+    public static class VictoryScreenPatch {
+        @SpirePostfixPatch
+        public static void logScoreBreakdown(VictoryScreen _instance, ArrayList<GameOverStat> ___stats) {
+            createScoreBreakdown(___stats);
+        }
+    }
+
+    @SpirePatch(clz = VictoryScreen.class, method = SpirePatch.CONSTRUCTOR)
+    public static class InsertEarlyStatCalcVictoryPatch {
+        @SpireInsertPatch(locator = Locator.class)
+        public static void insertEarlyVictoryStatCalcVictory(VictoryScreen __instance) {
+            // both methods are called twice now, but this shouldn't cause any problems
+            ReflectionHacks.privateMethod(GameOverScreen.class, "calculateUnlockProgress").invoke(__instance);
+            ReflectionHacks.privateMethod(DeathScreen.class, "createGameOverStats").invoke(__instance);
```

Comment:
This needs to be `VictoryScreen.class`, as written this will crash because the type listed here doesn't match the type of the object being passed in.

Make sure to test things out with beating the Heart -- if I remember the logic for these screens correctly, that's the only time VictoryScreen is called. Beating the act 3 bosses without going to act 4 still uses DeathScreen.

Reply:
Thanks for catching this c/p mistake. It's fixed and tested on a heart run now.